src/plugins/discover/public/application/main/components/document_explorer_callout/document_explorer_callout.tsx

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useState } from 'react';
+import './document_explorer_callout.scss';
+import { EuiButton, EuiButtonIcon, EuiCallOut, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { getServices } from '../../../../kibana_services';
+import { DOC_TABLE_LEGACY } from '../../../../../common';
+import { Storage } from '../../../../../../kibana_utils/public';
+
+const CALLOUT_STATE_KEY = 'discover:docExplorerCalloutClosed';
+
+const getStoredCalloutState = (storage: Storage): boolean => {
+  const calloutClosed = storage.get(CALLOUT_STATE_KEY);
+  if (typeof calloutClosed === 'boolean') {
+    return calloutClosed;
+  }
+  return false;
```

Comment:
you could make it a bit shorter and flexible by 
```suggestion
  return Boolean(calloutClosed);
```